UI/FleetButton.cpp

```diff
@@ -291,6 +313,40 @@ void FleetButton::LayoutIcons() {
         GG::Pt graphic_ul = middle - subtexture_sz / 2;
         m_selection_indicator->SizeMove(graphic_ul, graphic_ul + subtexture_sz);
     }
+
+    // refresh fleet button tooltip
+    if (m_fleet_blockaded) {
+        std::shared_ptr<Fleet> fleet;
+        std::shared_ptr<System> current_system;
+        std::map<int, bool> adjacent_system_IDs;
+        std::string available_exits = "";
+        int available_exits_count = 0;
+
+        if (!m_fleets.empty())
+            // can just pick first fleet because all fleets in system should have same exits
+            fleet = GetFleet(*m_fleets.begin());
```

Comment:
Sorry, I've tried to find the right place in the code, but it is pretty much beyond me. However, I have checked it in game, and non-monster agressive fleets do indeed share exits. I moved in on a hostile homeworld from to sides, and was able to retreat to either systems I was coming from, but not to a third adjacent system.